### PR TITLE
feat: add challenge 49

### LINKED ISSUE #713: New Challenge: use weak KDF to protect a secret
### Context

This is a Docker challenge focused on using the wrong KDF to protect a secret. 
In crypto-js there is an AES encryption mechanism, which uses MD5 as its KDF. This library is often used on mobile for encryption in hybrid apps. So what if we make a challenge in which the user has to find the right "pin"to be able to decrypt a secret offered on screen? (E.g. a 4-8 digit pin with md5 based KDF, and a secret fitting in 128 bytes.
We need to relate it to the MSTG on how to use (P)KDF with additional entropy and contextual binding.

### Did you encounter this in real life? Could you tell us more about the scenario?
See https://github.com/brix/crypto-js/blob/c8a2312474ae60c823f3c00b4d7aac2da460bbfc/test/config-test.js for test defaults.

### PR DESCRIPTION
**Pull request by adarsh-a-tw**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [x] A new challenge
-   [ ] Additional documentation
-   [ ] Something else

#### Description

Adds Challenge 49

### Relations

Closes #713 

### Checklist:

-   [x] All the contributions made are solely the work of me and my co-authors
-   [x] I tested the changes in this PR (if applicable)
-   [x] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] I added UI tests to ensure my UI changes work (when change in the overall UI, not needed if just adding a challenge)
-   [x] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge49.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import org.owasp.wrongsecrets.challenges.Challenge;
+import org.owasp.wrongsecrets.challenges.Spoiler;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/** This is a challenge based on using weak KDF to protect secrets. */
+@Slf4j
+@Component
+@Order(49)
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge49.java:23]:
```suggestion
@Order(49)
```
the order is no longer required :)

Comment (adarsh-a-tw) [reply src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge49.java]: Removed it

File: src/main/resources/explanations/challenge49_reason.adoc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+*Why Using MD5 as a KDF is Bad ?*
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge49_reason.adoc:1]:
```suggestion
*Why Using MD5 as a KDF is Bad*
```

File: src/main/resources/explanations/challenge49_reason.adoc [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+Protecting keys effectively is crucial, and this means using the right Key Derivation Functions (KDFs) with additional entropy and contextual binding, as emphasized in the https://mas.owasp.org/MASTG/0x04g-Testing-Cryptography/#weak-key-generation-functions[Mobile Security Testing Guide (MSTG).]
+
+MD5 is too fast and easy to compute, enabling attackers to quickly try a vast number of inputs (like PINs) to derive the key.
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge49_reason.adoc:5]:
```suggestion
MD5 is too fast and easy to compute, enabling attackers to quickly try a vast number of inputs (like PINs) to derive the key. Additionally (although a bit harder to exploit), the collision space is relatively small, meaning multiple different inputs can lead to the same md5 hash.
```

**Review by bendehaan — COMMENTED** [review]
Thanks for your contribution, nice addition! Some minor textual nits

**Pull request merged**